### PR TITLE
fix(passport): Fixing legal residence check

### PR DESCRIPTION
**Pull request by joningi98**
## What

Currently, our passport application system restricts access to users without legal residence, preventing them from applying for passports on behalf of their children. 

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205738665206036

### CODE REVIEW

**Review by kksteini — APPROVED** [review]
LGTM

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts [hunk 1]
```diff
@@ -31,11 +31,34 @@ export class NationalRegistryService extends BaseTemplateApiService {
     auth,
     params,
   }: TemplateApiModuleActionProps<NationalRegistryParameters>): Promise<NationalRegistryIndividual | null> {
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:33]: This functionality complicates this function somewhat. So I would suggest breaking this up like so. 
``` 
const result = await this.getIndividual(auth.nationalId);

if (!result) {
        throw new TemplateApiError(
            {
                title: coreErrorMessages.nationalIdNotFoundInNationalRegistryTitle,
                summary: coreErrorMessages.nationalIdNotFoundInNationalRegistrySummary,
            },
            400
        );
    }
    
    this.validateIndividual(result, params);
    
    if (params?.includeChildrenInValidation) {
        const children = await this.nationalRegistryApi.getCustodyChildren(auth);
        await this.validateChildren(children, params);
    }

    return result;

``` 
Where validateChildren loops through each child uses the validateIndividual and  then a separate function for each validation.

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts [hunk 2]
```diff
@@ -31,11 +31,34 @@ export class NationalRegistryService extends BaseTemplateApiService {
...
+    const individuals: Array<NationalRegistryIndividual | null> = []
     const result = await this.getIndividual(auth.nationalId)
+    individuals.push(result)
+
+    if (params?.allowDomicilePassOnChild) {
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:38]: Are we not checking for more than just domicile?  `includeChildrenInValidation`  as suggestion ?

**Review by obmagnusson — CHANGES_REQUESTED** [review]
Sry for the late reply. Couple of comments.

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts (hunk 1 above)
Comment (joningi98) [reply libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:32]: I've broken up the function, makes the validation part easier to read

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts [hunk 3]
```diff
@@ -31,39 +30,100 @@ export class NationalRegistryService extends BaseTemplateApiService {
...
+    return individual
+  }
+
+  private validateIndividual(
+    individual: NationalRegistryIndividual,
+    isChild: boolean,
+    params?: NationalRegistryParameters,
+  ) {
+    if (params?.legalDomicileIceland && !params?.allowPassOnChild) {
+      this.validateDomicleInIceland(individual, params.allowPassOnChild)
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:65]: Won't `params.allowPassOnChild` always be `false` or `undefined` here?
Is that expected?

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts [hunk 4]
```diff
@@ -31,39 +30,100 @@ export class NationalRegistryService extends BaseTemplateApiService {
...
+
+    if (params?.ageToValidate && !isChild) {
+      this.validateAge(params, individual)
+    }
+  }
+
+  private async validateChildren(params: NationalRegistryParameters, childrenId: string[]) {
+    for (const id of childrenId) {
+      const individual = await this.getIndividual(id)
+      console.log('validateChildren', individual)
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:83]: Shuold this console log be here?

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts (hunk 3 above)
Comment (joningi98) [reply libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:65]: your right!

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts [hunk 5]
```diff
@@ -31,39 +30,91 @@ export class NationalRegistryService extends BaseTemplateApiService {
...
+        {
+          title: coreErrorMessages.nationalRegistryCitizenshipNotIcelandic,
+          summary: coreErrorMessages.nationalRegistryCitizenshipNotIcelandic,
+        },
+        400,
+      )
+    }
+  }
+
+  private validateAge(params: any, individual: NationalRegistryIndividual) {
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:116]: params: NationalRegistryParameters

File: libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts [hunk 6]
```diff
@@ -31,39 +30,91 @@ export class NationalRegistryService extends BaseTemplateApiService {
...
+  ) {
+    for (const id of childrenId) {
+      const individual = await this.getIndividual(id)
+      if (individual) {
+        this.validateIndividual(individual, true, params)
       }
     }
+  }
+
+  private validateDomicleInIceland(individual: NationalRegistryIndividual) {
```
Comment (obmagnusson) [inline libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts:89]: Typo : validateDomicleInIceland -> validateDomicileInIceland

**Review by obmagnusson — CHANGES_REQUESTED** [review]
Nice ! Minor comments. Has this been tested on other applications?

**Review by obmagnusson — APPROVED** [review]

**Pull request merged**